test/es6/ES6Super.js

```diff
@@ -0,0 +1,202 @@
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+  {
+    name: "super in direct eval inside an object method",
+    body: function () {
+        var value;
+        var obj = {
+            m() { value = eval('super.value;'); }
+        };
+
+        assert.areEqual(undefined, value);
+        obj.m();
+        assert.areEqual(undefined, value);
+        Object.setPrototypeOf(obj, { value: "super" });
+        obj.m();
+        assert.areEqual("super", value);
+    }
+  },
+  {
+    name: "home object's prototype is null",
+    body: function () {
+        var value;
+        var obj = {
+            m() {
+                value = "identifier";
+                super.prop;
+             },
+            n() {
+                value = "expression";
+                super["prop"];
+            }
+        };
+        
+        Object.setPrototypeOf(obj, null);
+        assert.throws(obj.m, TypeError, "identifier", "Unable to get property 'prop' of undefined or null reference");
+        assert.throws(obj.n, TypeError, "expression", "Unable to get property 'prop' of undefined or null reference");
+    }
+  },
+  {
+    name: "super in strict mode - object",
+    body: function () {
+        "use strict";
+        var obj0 = {
+            m() {
+                super.prop = "identifier";
+                Object.freeze(obj0);
+                try {
+                    super.prop = "super";
+                    throw Error("Expected to throw");
+                } catch (e) {
+                    assert.areEqual("TypeError", e.name);
+                    assert.areEqual("Assignment to read-only properties is not allowed in strict mode", e.description);
+                }
+            }
+        };
+        
+        var obj1 = {
+            m() {
+                super['prop'] = "expression";
+                Object.freeze(obj1);
+                try {
+                    super['prop'] = "super";
+                    throw Error("Expected to throw");
+                } catch (e) {
+                    assert.areEqual("TypeError", e.name);
+                    assert.areEqual("Assignment to read-only properties is not allowed in strict mode", e.description);
+                }
+            }
+        };
+        
+        var base = {};
+        Object.setPrototypeOf(obj0, base);
+        Object.setPrototypeOf(obj1, base);
+        obj0.m();
+        assert.areEqual("identifier", obj0.prop);
+        obj1.m();
+        assert.areEqual("expression", obj1.prop);
+    }
+  },
+  {
+    name: "super in strict mode - class",
+    body: function () {
+        class A {
+            m() {
+                super.prop = "identifier";
+                Object.freeze(A.prototype);
+                try {
+                    super.prop = "super";

```

Comment:
Updated in latest rev.

---
In reply to: [75529926](https://github.com/Microsoft/ChakraCore/pull/1466#discussion_r75529926) [](ancestors = 75529926)